src/test/java/com/hartwig/snpcheck/SnpCheckTest.java

```diff
@@ -214,74 +232,139 @@ public void finishedSingleSampleRunComparedToValidationVcfPass() {
         assertThat(update.getStatus()).isEqualTo(Status.VALIDATED);
     }
 
-    @SuppressWarnings("unchecked")
     @Test
     public void publishesTurquoiseEventOnCompletion() {
         fullSnpcheckWithResult(VcfComparison.Result.PASS, BARCODE);
         victim.handle(stagedEvent(Context.DIAGNOSTIC));
-        ArgumentCaptor<PubsubMessage> pubsubMessageArgumentCaptor = ArgumentCaptor.forClass(PubsubMessage.class);
-        verify(turquoiseTopicPublisher, times(1)).publish(pubsubMessageArgumentCaptor.capture());
-        Map<String, Object> event = readEvent(pubsubMessageArgumentCaptor, Map.class);
-        assertThat(event.get("type")).isEqualTo("snpcheck.completed");
-        List<Map<String, Object>> subjects = (List<Map<String, Object>>) event.get("subjects");
-        assertThat(subjects.get(0).get("name")).isEqualTo("samplet");
-        assertThat(subjects.get(0).get("type")).isEqualTo("sample");
+        var turquoiseEvents = eventBuilder.getQueueBuffer(new TurquoiseEvent.EventDescriptor());
+        assertThat(turquoiseEvents).hasSize(1);
+        var event = turquoiseEvents.get(0);
+        assertThat(event.type()).isEqualTo("snpcheck.completed");
+        assertThat(event.subjects().get(0).name()).isEqualTo("samplet");
+        assertThat(event.subjects().get(0).type()).isEqualTo("sample");
+    }
+
+    @Test
+    public void publishesAquaEventOnDiagnosticCompletion() {
+        fullSnpcheckWithResult(VcfComparison.Result.PASS, BARCODE);
+        victim.handle(stagedEvent(Context.DIAGNOSTIC));
+        var aquaEvents = eventBuilder.getQueueBuffer(new AquaEvent.EventDescriptor());
+        assertThat(aquaEvents).hasSize(1);
+        var event = (SnpCheckCompletedEvent) aquaEvents.get(0);
+        assertThat(event.type()).isEqualTo(AquaEventType.SNP_CHECK_COMPLETED);
+        assertThat(event.barcode()).isEqualTo(BARCODE);
+        assertThat(event.snpCheckResult()).isEqualTo("PASS");
+        assertThat(event.setName()).isEqualTo("set");
+        assertThat(event.context()).isEqualTo(Context.DIAGNOSTIC);
     }
```

Comment:
Maybe good to also add a test case were a run only has a reference sample, to test the fallback to the reference barcode in the Aqua event.